### DISCUSSION TITLE
`allOf` rendering does not seem correct

### DISCUSSION THREAD

**Discussion by sircodemane** [scalar/scalar#2745] [Q&A]
I originally intended to open a bug report for this, but it seemed more appropriate to discuss first. `allOf` seems to render in a similar manner as `anyOf` and `oneOf`, but I don't believe that is how to correctly represent `allOf` schemas. For `anyOf` and `oneOf`, it makes sense to render each element separately as each element is considered a bespoke model. However, the elements of `allOf` should not be treated or rendered in this manner; instead the elements should be merged together to create a single combined model. I'd be curious to hear others thoughts as different toolings seem to have different opinions on this.

**Comment by marclave** [comment]
@hanspagel thoughts?

i think this is a reasonable (and logical) way of rendering !

**Comment by CoolGoose** [comment]
Any updates here ? :) We also kinda use a lot of allof and it looks ugly as hell :D
